sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -386,7 +386,10 @@ static void applyOptionsFromImage(std::string &CompileOpts,
   if (isEsimdImage && pi::DeviceBinaryProperty(isEsimdImage).asUint32()) {
     if (!CompileOpts.empty())
       CompileOpts += " ";
-    CompileOpts += "-vc-codegen";
+    // TODO: "-vc-enable-preemption" is added temporarily until the option
+    // is enabled by default in vc-codegen. That has not been done yet due to
+    // still unresolved issues with non-sycl users of vc-backend.
+    CompileOpts += "-vc-codegen -vc-enable-preemption";
```

Comment:
From @kychendev explanation, this flag may hurt performance as it reduces the number of registers available to RA. I understand that this fixes correctness in some cases, but it also lowers performance for other cases which don't suffer from the correctness problem. Is there a specific pattern in the code when preemption is needed to enforce correctness? I'd vote for this being documented and providing the user with way to opt-out.
adding @kvladimi